server/sonar-server/src/main/java/org/sonar/server/app/TomcatConnectors.java

```diff
@@ -48,11 +47,16 @@ static void configure(Tomcat tomcat, Props props) {
   private static Connector newHttpConnector(Props props) {
     // Not named "sonar.web.http.port" to keep backward-compatibility
     int port = props.valueAsInt("sonar.web.port", 9000);
-    if (port < UNDEFINED_PORT) {
+    if (port < 0) {
       throw new IllegalStateException(format("HTTP port '%s' is invalid", port));
     }
 
-    Connector connector = newConnector(props, HTTP_PROTOCOL, "http");
+    Connector connector = new Connector(HTTP_PROTOCOL);
+    connector.setURIEncoding("UTF-8");
+    connector.setProperty("address", props.value("sonar.web.host", "0.0.0.0"));
```

Comment:
This is the default value, as commented in sonar.properties. Users can use IPv6 host.